src/main/kotlin/cacophony/codegen/linearization/Linearization.kt

```diff
@@ -1,10 +1,128 @@
+    var blocks = 0
+
+    fun getLabel() = "bb${blocks++}"
+
+    fun dfs(label: CFGLabel): PartialLinearization {
+        val vertex = fragment.vertices[label]!!
+        val block = MutableBasicBlock(BlockLabel(getLabel()))
+        visited[label] = block
+
+        fun handle(vertex: CFGVertex.Final) =
+            Pair(
+                block.also { it.instructions += covering.coverWithInstructions(vertex.tree) },
+                iterator { yield(block) },
+            )
+
+        fun handle(vertex: CFGVertex.Jump): PartialLinearization {
+            var result = iterator<MutableBasicBlock> { yield(block) }
+
+            (
+                visited[vertex.destination] ?: dfs(vertex.destination).let { (neighbor, iter) ->
+                    result = combine(result, iter)
+                    neighbor
+                }
+            ).also { neighbor ->
+                block.successors.add(neighbor)
+                neighbor.predecessors.add(block)
+                block.instructions += covering.coverWithInstructionsAndJump(vertex.tree, neighbor.label)
+            }
+
+            return Pair(block, result)
+        }
+
+        fun handle(vertex: CFGVertex.Conditional): PartialLinearization {
+            var result = iterator<MutableBasicBlock> { yield(block) }
+
+            var swapped = false
+            var doubled = false
+
+            val falseBlock =
+                (
+                    visited[vertex.falseDestination]?.also { swapped = true } ?: dfs(vertex.falseDestination).let { (neighbor, iter) ->
+                        result = combine(result, iter)
+                        neighbor
+                    }
+                ).also { neighbor ->
+                    block.successors.add(neighbor)
+                    neighbor.predecessors.add(block)
+                }
+
+            val trueBlock =
+                (
+                    visited[vertex.trueDestination]?.also {
+                        doubled = swapped
+                        swapped = false
+                    }
+                        ?: dfs(vertex.trueDestination).let { (neighbor, iter) ->
+                            result = combine(result, iter)
+                            neighbor
+                        }
+                ).also { neighbor ->
+                    block.successors.add(neighbor)
+                    neighbor.predecessors.add(block)
+                }
+
+            if (swapped) {
+                // false edge goes backwards, true edge forwards
+                block.instructions += covering.coverWithInstructionsAndJump(CFGNode.LogicalNot(vertex.tree), falseBlock.label)
+            } else if (doubled) {
+                // both edges go backwards, need 2 jumps
+                block.instructions += covering.coverWithInstructionsAndJump(vertex.tree, trueBlock.label)
+                block.instructions += covering.coverWithInstructionsAndJump(CFGNode.Constant(42), falseBlock.label)
```

Comment:
**question:** What is the semantics of `coverWithInstructionsAndJump`? How to force 2 jumps in this scenario?